facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -94,6 +105,21 @@ public ServiceFacade(D legacySvc, D lightblueSvc, String implementationName, Pro
 
         timeoutConfiguration = new TimeoutConfiguration(DEFAULT_TIMEOUT_MS, implementationName, this.properties);
 
+        // create ThreadPoolExecutor for all facades
+        synchronized("executor") {
+            if (executor == null) {
+                int threadPoolSize = Integer.parseInt(this.properties.getProperty(CONFIG_PREFIX+implementationName+".threadPool.size", "50"));
+
+                // cached means automatic thread reclamation
+                ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) Executors.newCachedThreadPool();
+                // fixed size, no queuing
+                threadPoolExecutor.setCorePoolSize(threadPoolSize);
+                threadPoolExecutor.setMaximumPoolSize(threadPoolSize);
+                executor = MoreExecutors.listeningDecorator(threadPoolExecutor);
```

Comment:
For the null checks around synchronization to work correctly save the created executor into a local variable first then assign to `executor`.  Used to be (aka maybe not a problem anymore) null check could see a field as not null while the assigned object was still being constructed.